content/gsoc/2024.md

```diff
@@ -0,0 +1,606 @@
+
+### Tasks
+
+- Implement a RZIL uplifting for any non-trivial architecture, preferably that is supported by ESIL already
+- Improve the integration with analysis (variables and types differences) for the chosen architecture
+- Write the test cases for Rizin regression tests and improve the results.
+- Update and use [rz-tracetest](https://github.com/rizinorg/rz-tracetest) for the chosen architectures
+- Implement necessary commands and APIs in Rizin for visual representation of the IL tree
+- Implement standard and graph views in Cutter for the IL output (optional)
+
+Due to the sensivity of uplifting to the precision, it's important to follow these steps:
+- For every single lifted opcode, have at the very least one asm test in `test/db/asm/...` containing the IL to detect
+changes when the code is changed and have it type-checked. This should produce 100% coverage on the lifter C code (except e.g. `malloc()` error handling).
+- Run `rz-tracetest` on real traces. It's also possible to write custom assembly programs that execute specific obscure
+instructions where it's hard to be sure that they were implemented correctly on many random inputs and then feed these executions into `rz-tracetest`.
+- Few `rz-test` command tests that emulate some code snippets in rizin. For example a simple decryption loop to check
+the overall integration in rizin, or for specific edge cases (like running a division by zero).
+
+### Skills
+The participant should know C and bits of C++ as well as be familiar with basics of the program analysis. Having an
+experience with other intermediate language, SAT/SMT, and mathematical logic is a plus.
+
+### Difficulty
+Medium
+
+### Benefits for the participant
+The participant will understand the state of the art of intermediate languages research, it's relation to the mathematical logic, SMT, and program analysis.
+Moreover, the participant will become familiar with the both symbolic and concrete emulation during the
+implementation process.
+
+### Benefits for the project
+Migrating most architectures will help to deprecate and remove outdated ESIL and will help improving the analysis precision.
+Adding uplifting for new architectures that weren't even supported by ESIL will improve the
+analysis to even greater degree.
+
+### Assess requirements for midterm/final evaluation
+- 1st term: finish the RZIL uplifting for the chosen architecture with basic instruction-level tests
+- Final term: implement all changes in the analysis code, added more complex integration tests with types analysis
+
+### Mentors
+- xvilka
+- thestr4ng3r
+
+### Links/Resources
+- [RZIL-labeled issues](https://github.com/rizinorg/rizin/labels/RZIL)
+- [ESIL-labeled issues](https://github.com/rizinorg/rizin/labels/RZIL)
+- [ESIL to RZIL conversion tracking issue](https://github.com/rizinorg/rizin/issues/2080)
+- [Cutter: IL output and graph visual representation](https://github.com/rizinorg/rizin/issues/1133)
+
+## FRIDA Integration
```

Comment:
there are two `Frida integration`. we should change the name of one of these.

Reply:
Not sure what you are talking about? It has only one.

Reply:
```suggestion
```

Reply:
i think this was something that was moved further down and this specific section remained.